sr3 downloads fail when the downloaded size doesn't match the size in the message

In some cases, the code reaches this call: https://github.com/MetPX/sarracenia/blob/459cc4e726e5a84d76db9f1930a3634d1456f966/sarracenia/flow/__init__.py#L1890

when `new_file` does not exist because the inflight file hasn't been renamed to new_file.

```log
2023-06-26 18:24:40,024 [WARNING] sarracenia.flow do_download downloading again, attempt 3
2023-06-26 18:24:40,024 [DEBUG] sarracenia.flow download https_transport download relPath=/outgoing/OMTO3/OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5
2023-06-26 18:24:40,024 [DEBUG] sarracenia.flow download baseUrl.path= /
 --
2023-06-26 18:24:40,026 [DEBUG] sarracenia.transfer.https __open__ bearer_token: c3NjaHBjOmMzTmpMbWh3WXkxamFIQXVjM0JqUUdOaGJtRmtZUzVqWVE9PToxNjQ3Mzc0MjMyOmU2M2RiMDY5YWI3ODA1N2QwZTBmYTBiMTNhOWExYjkyYTg5NjI5ZmI
2023-06-26 18:24:40,174 [DEBUG] sarracenia.transfer read_write sr_proto read_write
2023-06-26 18:24:43,684 [WARNING] sarracenia.flow download downloaded 33214240 of with no length given for OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5.tmp assuming ok
2023-06-26 18:24:43,684 [DEBUG] sarracenia.flow download Exception details:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 1871, in download
    self.set_local_file_attributes(new_file, msg)
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 2263, in set_local_file_attributes
    x.set('mtime', sarracenia.timeflt2str(os.path.getmtime(local_file)))
  File "/usr/lib/python3.6/genericpath.py", line 55, in getmtime
    return os.stat(filename).st_mtime
 FileNotFoundError: [Errno 2] No such file or directory: 'OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5'
 2023-06-26 18:24:43,685 [WARNING] sarracenia.flow download failed to write OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5.tmp: [Errno 2] No such file or directory: 'OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5'
2023-06-26 18:24:43,685 [DEBUG] sarracenia.transfer.https close sr_http close
2023-06-26 18:24:43,685 [DEBUG] sarracenia.transfer.https init sr_http init
2023-06-26 18:24:43,688 [INFO] sarracenia.flow do_download attempt 3 failed to download https://omisips1.omisips.eosdis.nasa.gov///outgoing/OMTO3/OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5 to /apps/sarra/public_data/20230626/NASA/OMTO3/OMI-Aura_L2-OMTO3_2023m0626t1352-o100778_v883-2023m0626t165246.he5
2023-06-26 18:24:43,688 [ERROR] sarracenia.flow do_download gave up downloading for now, appending to retry queue
2023-06-26 18:24:43,688 [DEBUG] sarracenia.flow.sarra do processing 0 messages worked!
```

This is the relevant section of code where renaming _sometimes_ occurs:

https://github.com/MetPX/sarracenia/blob/459cc4e726e5a84d76db9f1930a3634d1456f966/sarracenia/flow/__init__.py#L1821-L1869

I think the issue is that the file is only renamed from new_inflight_path to new_file when len_written == block_length. But there's three (of four) other cases where it should also rename the file - every case except where False is returned:

https://github.com/MetPX/sarracenia/blob/459cc4e726e5a84d76db9f1930a3634d1456f966/sarracenia/flow/__init__.py#L1848-L1867

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1835,13 +1835,6 @@ class Flow:
                 if not self.o.dry_run:
                     if accelerated:
                         self.proto[self.scheme].update_file(new_inflight_path)
-                    if (new_inflight_path != new_file):
-                        if os.path.isfile(new_file):
-                            os.remove(new_file)
-                        os.rename(new_inflight_path, new_file)
-                    # older versions don't include the contentType, so patch it here.
-                    if 'contentType' not in msg:
-                        msg['contentType'] = magic.from_file(new_file,mime=True)
             elif len_written < 0:
                 logger.error("failed to download %s" % new_file)
                 return False
@@ -1865,8 +1858,18 @@ class Flow:
                             'incomplete download only %d of expected %d bytes for %s'
                             % (len_written, block_length, new_inflight_path))
                         return False
+                # when len_written is different than block_length
+                msg['size'] = len_written
 
-                    msg['size'] = len_written
+            # if we haven't returned False by this point, assuming download was successful
+            if (new_inflight_path != new_file):
+                if os.path.isfile(new_file):
+                    os.remove(new_file)
+                os.rename(new_inflight_path, new_file)
+            
+            # older versions don't include the contentType, so patch it here.
+            if 'contentType' not in msg:
+                msg['contentType'] = magic.from_file(new_file,mime=True)
 
             self.metrics['flow']['transferRxBytes'] += len_written
             self.metrics['flow']['transferRxFiles'] += 1

